score-xml/src/main/java/io/cloudslang/content/xml/actions/EditXml.java

```diff
@@ -0,0 +1,191 @@
+
+            switch (myAction) {
+                case delete: {
+                    result.put(Constants.Outputs.RETURN_RESULT, XmlOperationsUtils.delete(xml, filePath, xpath1, type, name, features));
+                    break;
+                }
+                case insert: {
+                    result.put(Constants.Outputs.RETURN_RESULT, XmlOperationsUtils.insert(xml, filePath, xpath1, value, type, name, features));
+                    break;
+                }
+                case append: {
+                    result.put(Constants.Outputs.RETURN_RESULT, XmlOperationsUtils.append(xml, filePath, xpath1, value, type, name, features));
+                    break;
+                }
+                case subnode: {
+                    result.put(Constants.Outputs.RETURN_RESULT, XmlOperationsUtils.createSubnode(xml, filePath, xpath1, value, features));
+                    break;
+                }
+                case move: {
+                    result.put(Constants.Outputs.RETURN_RESULT, XmlOperationsUtils.move(xml, filePath, xpath1, xpath2, features));
+                    break;
+                }
+                case rename: {
+                    result.put(Constants.Outputs.RETURN_RESULT, XmlOperationsUtils.rename(xml, filePath, xpath1, name, type, value, features));
+                    break;
+                }
+                case update: {
+                    result.put(Constants.Outputs.RETURN_RESULT, XmlOperationsUtils.update(xml, filePath, xpath1, value, type, name, features));
+                    break;
+                }
+            }
+            result.put(Constants.Outputs.RETURN_CODE, String.valueOf(Constants.ReturnCodes.SUCCESS));
+        } catch (IllegalArgumentException e) {
+            result.put(Constants.Outputs.EXCEPTION, "Invalid action " + action);
+            result.put(Constants.Outputs.RETURN_CODE, String.valueOf(Constants.ReturnCodes.FAILURE));
+        } catch (Exception e) {
+            result.put(Constants.Outputs.EXCEPTION, e.getMessage());
+            result.put(Constants.Outputs.RETURN_CODE, String.valueOf(Constants.ReturnCodes.FAILURE));
+        }
+        return result;
+    }
+
+    private ActionsEnum getAction(String action) {
+        return ActionsEnum.valueOf(action.toLowerCase());
+    }
+
+    /**
+     * Validates the operation inputs.
+     *
+     * @param xml      the xml as String
+     * @param filePath the path/remote path to the file
+     * @param action   the edit action
+     * @param xpath1   the first XPath
+     * @param xpath2   the second XPath
+     * @param type     the type on which the edit should occur
+     * @param name     the name of the attribute
+     * @throws Exception for invalid inputs
+     */
+    private void validateInputs(String xml, String filePath, String action, String xpath1, String xpath2, String type, String name, String features) throws Exception {
+        ValidateUtils.validateXmlAndFilePathInputs(xml, filePath);
```

Comment:
Isn't it possible to reuse this method for other XML operations if necessary, and extract it in a different class?